### PR TITLE
[DeviceSanitizer] Support detecting out-of-bounds error on DeviceGlobals

### PR DESCRIPTION
**Pull request by zhaomaosu**
*Fix event lists are not correctly passed to KernelLaunch

intel/llvm PR: https://github.com/intel/llvm/pull/12753

### CODE REVIEW

File: source/loader/layers/sanitizer/asan_interceptor.hpp [hunk 1]
```diff
@@ -163,4 +178,21 @@ class SanitizerInterceptor {
     bool m_ShadowMemInited;
 };
 
+inline const char *ToString(AllocType Type) {
```
Comment (wenju-he) [inline source/loader/layers/sanitizer/asan_interceptor.hpp:181]:
```suggestion
inline const char *AllocTypeToString(AllocType Type) {
```

**Review by wenju-he — APPROVED** [review]
LGTM

**Review by pbalcer — APPROVED** [review]

**Review by AllanZyne — APPROVED** [review]
LGTM

**Comment by pbalcer** [comment]
please rebase and update the intel/llvm PR.

**Comment by zhaomaosu** [comment]
> please rebase and update the intel/llvm PR.

Done. The pre-ci of intel/llvm PR has finished, there is one windows failure which is known issue and is tracked by https://github.com/intel/llvm/issues/12798. @pbalcer, could you please help merge this PR?

**Pull request merged**